### PR TITLE
socketconnection: add PMI socket to blacklisted sockets

### PR DESCRIPTION
**Pull request by rohgarg**
This is a workaround for the random ckpt bug at Stampede.

### CODE REVIEW

File: src/plugin/ipc/socket/socketconnection.cpp [hunk 1]
```diff
@@ -221,6 +235,15 @@ bool TcpConnection::isBlacklistedTcp(const sockaddr* saddr, socklen_t len)
     int blacklistedRemotePorts[] = {53,                 // DNS Server
                                     389, 636,           // LDAP
                                     -1};
+#ifdef MPISPAWN_WORKAROUND
```
Comment (karya0) [inline src/plugin/ipc/socket/socketconnection.cpp]: May be a more readable design here would be to replace the int array with a vector and then just do a push_back to the vector inside this #ifdef. What do you think?

**Comment by karya0** [comment]
LGTM modulo one small style comment. Feel free to ignore it :).

File: src/plugin/ipc/socket/socketconnection.cpp (hunk 1 above)
Comment (rohgarg) [inline src/plugin/ipc/socket/socketconnection.cpp]: Sure. That sounds better.

Comment (rohgarg) [inline src/plugin/ipc/socket/socketconnection.cpp]: Thinking again, it'll be a performance overhead unless we can move it to a global constructor. The cost is minor but can quickly add up for an application that uses many sockets.

I'll keep it as is for now.

**Pull request merged**